pycsw/ogc/gml/gml3.py

```diff
@@ -47,6 +47,11 @@ def _poslist2wkt(poslist, axisorder):
     tmp = poslist.split()
     poslist2 = []
 
+    if len(tmp) < 4:
+        msg = 'Invalid number of coordinates in geometry'
+        LOGGER.error(msg)
+        raise RuntimeError(msg)
+
     xlist = tmp[1::2]
     ylist = tmp[::2]
```

Comment:
Will you change those two lines in another PR?

Reply:
The output behaviour:

```python
>>> [1, 2, 3, 4, 5, 6][1::2]
[2, 4, 6]
>>> [1, 2, 3, 4, 5, 6][::2]
[1, 3, 5]
```

Reply:
Yes, but as already described in https://github.com/geopython/pycsw/issues/621#issue-665176298, the above expression creates inconsistencies with respect to a GML point. The following example with the point `40 10` (latitude, longitude/ X, Y) and the (invalid) polygon `40 10` (latitude, longitude/ X, Y) should illustrate this:

For a GML point, you select the coordinates as follows: 

```
xypoint = tmp.text.split()
if self.crs.axisorder == 'yx':
    self.wkt = 'POINT(%s %s)' % (xypoint[1], xypoint[0])
else:
    self.wkt = 'POINT(%s %s)' % (xypoint[0], xypoint[1]) 
```
Assuming default order ("xy") the coordinates are selected accordingly, `40` as latitude, `10` as longitude.
However, for the Polygon its different:

```
xlist = tmp[1::2]
ylist = tmp[::2]

if axisorder == 'yx':
	for i, j in zip(ylist, xlist):
		poslist2.append('%s %s' % (i, j))
else:
	for i, j in zip(xlist, ylist):
		poslist2.append('%s %s' % (i, j))
```
Assuming default order ("xy") the coordinates are selected in the wrong order, namely, `40` as longitude, `10` as latitude, because: 
```
>>>tmp_list = [40, 10]
>>>tmp_list[1::2]  # xlist
[10]
>>>tmp_list[::2] # ylist
[40]
```
but: 
```
>>>tmp_list = [40, 10]
>>>tmp_list[::2]  # xlist
[40]
>>>tmp_list[1::2] # ylist
[10]
```
would do the right job.

Reply:
Fixed. I've added a test (https://github.com/tomkralidis/pycsw/blob/issue-621/tests/functionaltests/suites/default/post/GetRecords-filter-bbox-poslist.xml) which yields the same results as https://github.com/tomkralidis/pycsw/blob/issue-621/tests/functionaltests/suites/default/post/GetRecords-filter-bbox.xml.

Reply:
Great, that's how I have expected it to work.